packages/web/src/components/typography/Text.tsx

```diff
@@ -0,0 +1,87 @@
+// import cn from 'classnames'
+
+import { CSSProperties, ReactNode } from 'react'
+
+import { getCurrentThemeColors, ThemeColor } from 'utils/theme/theme'
+
+import { variantInfoMap, variantStrengthMap } from './constants'
+import {
+  TextSizeInfo,
+  TextStrengthInfo,
+  TextVariant,
+  TextSize,
+  TextStrength,
+  FontWeight
+} from './types'
+
+export const fontWeightMap: Record<FontWeight, number> = {
+  heavy: 900,
+  bold: 700,
+  demiBold: 600,
+  medium: 500,
+  regular: 400,
+  light: 300,
+  thin: 200,
+  ultraLight: 100
+}
+
+const defaultBodyInfo: TextSizeInfo & TextStrengthInfo = {
+  tag: 'p',
+  fontSize: '16px',
+  lineHeight: '20px',
+  fontWeight: 'medium',
+  textTransform: 'inherit'
+}
+
+export type TextProps = {
+  className?: string
+  children?: ReactNode
+  variant?: TextVariant
+  size?: TextSize
+  strength?: TextStrength
+  color?: ThemeColor
+}
+
+export const Text = ({
+  className,
```

Comment:
a standard id like us to start using is to have the word `props` here instead of destructured, and then desturcture in the body. if you feel strongly not to do this, totally get it, but it's a stylistic preference that i like a lot.